frontend/src/metabase/dashboard/reducers.js

```diff
@@ -142,12 +143,15 @@ const dashboards = handleActions(
         ),
     },
     [Dashboards.actionTypes.UPDATE]: {
-      next: (state, { payload }) =>
-        assocIn(
-          state,
-          [payload.dashboard.id, "collection_id"],
-          payload.dashboard.collection_id,
-        ),
+      next: (state, { payload }) => {
+        return produce(state, draftState => {
+          const draftDashboard = draftState[payload.dashboard.id];
+          if (draftDashboard) {
+            draftDashboard.collection_id = payload.dashboard.collection_id;
+            draftDashboard.collection = payload.dashboard.collection;
```

Comment:
This is working slightly differently than before. Before it's possible that `state[payload.dashboard.id]` would be null, the previous reducer will modify a state to look something like this.
```ts
{
  1: {
    collection_id: 2
   }
}
```

Which on its own, doesn't really do anything as when visiting the dashboard this ID would be replaced by the response from the dashboard endpoints anyway.

The new approach is only updating the relevant properties when there are dashboard states.